src/main/java/frc/team2412/robot/subsystem/IntakeSubsystem.java

```diff
@@ -1,142 +1,214 @@
+            colorMatcher.addColorMatch(BLUE_CARGO_COLOR);
+        }
+
+        intakeSolenoidState = RETRACT;
+        intakeMotorState = STOPPED;
 
         intakeRetract();
         intakeStop();
 
         setName("IntakeSubsystem");
     }
 
-    // Methods.
+    // Methods
 
+    /**
+     * Spins motors inwards and updates motor state
+     */
     public void intakeIn() {
-        if (intakeSolenoidState == IntakeSolenoidState.EXTENDED) {
-            motorOuterAxle.set(IntakeConstants.INTAKE_IN_SPEED);
-            motorInnerAxle.set(IntakeConstants.INTAKE_IN_SPEED);
+        if (intakeSolenoidState == EXTEND) {
+            motorOuterAxle.set(INTAKE_IN_SPEED);
+            motorInnerAxle.set(INTAKE_IN_SPEED);
 
-            intakeMotorState = IntakeMotorState.IN;
+            intakeMotorState = IN;
         }
     }
 
+    /**
+     * Spins motors outwards and updates motor state
+     */
     public void intakeOut() {
-        if (intakeSolenoidState == IntakeSolenoidState.EXTENDED) {
-            motorOuterAxle.set(IntakeConstants.INTAKE_OUT_SPEED);
-            motorInnerAxle.set(IntakeConstants.INTAKE_OUT_SPEED);
+        if (intakeSolenoidState == IntakeSolenoidState.EXTEND) {
+            motorOuterAxle.set(INTAKE_OUT_SPEED);
+            motorInnerAxle.set(INTAKE_OUT_SPEED);
 
-            intakeMotorState = IntakeMotorState.OUT;
+            intakeMotorState = OUT;
         }
     }
 
+    /**
+     * Stops motors and updates motor state
+     */
     public void intakeStop() {
         motorOuterAxle.set(0);
         motorInnerAxle.set(0);
 
-        intakeMotorState = IntakeMotorState.STOPPED;
+        intakeMotorState = STOPPED;
     }
 
+    /**
+     * Extends solenoid and updates solenoid state
+     */
     public void intakeExtend() {
-        intakeSolenoidState = IntakeSolenoidState.EXTENDED;
+        intakeSolenoidState = EXTEND;
 
-        solenoid.set(IntakeState.EXTEND.value);
+        solenoid.set(EXTEND.value);
     }
 
+    /**
+     * Retracts solenoid and updates solenoid state
+     */
     public void intakeRetract() {
-        intakeSolenoidState = IntakeSolenoidState.RETRACTED;
+        intakeSolenoidState = RETRACT;
+
+        solenoid.set(RETRACT.value);
+    }
 
-        solenoid.set(IntakeState.RETRACT.value);
+    /**
+     * Returns true if color sensor detects an enemy cargo
+     */
+    public boolean individualHasOpposingColorCargo(MultiplexedColorSensor colorSensor) {
+        return colorMatcher.matchColor(colorSensor.getColor()) != null;
     }
 
-    public boolean colorIdentify() {
-        return true;
+    /**
+     * Returns true if any of the color sensors detect an enemy cargo
+     */
+    public boolean hasOpposingColorCargo() {
+        if (individualHasOpposingColorCargo(leftColorSensor) || individualHasOpposingColorCargo(rightColorSensor)
+                || individualHasOpposingColorCargo(centerColorSensor)) {
+            return true;
+        }
+        return false;
```

Comment:
I would recommend removing the if statement and returning the condition to make things cleaner.